back/src/domain/convention/useCases/notifications/NotifyAllActorsThatConventionIsDeprecated.ts

```diff
@@ -0,0 +1,68 @@
+import { ConventionDto, conventionSchema } from "shared";
+import {
+  UnitOfWork,
+  UnitOfWorkPerformer,
+} from "../../../core/ports/UnitOfWork";
+import { TransactionalUseCase } from "../../../core/UseCase";
+import { SaveNotificationAndRelatedEvent } from "../../../generic/notifications/entities/Notification";
+
+export class NotifyAllActorsThatConventionIsDeprecated extends TransactionalUseCase<ConventionDto> {
+  constructor(
+    uowPerformer: UnitOfWorkPerformer,
+    private readonly saveNotificationAndRelatedEvent: SaveNotificationAndRelatedEvent,
+  ) {
+    super(uowPerformer);
+  }
+
+  inputSchema = conventionSchema;
+
+  public async _execute(
+    convention: ConventionDto,
+    uow: UnitOfWork,
+  ): Promise<void> {
+    const [agency] = await uow.agencyRepository.getByIds([convention.agencyId]);
+    if (!agency) {
+      throw new Error(
+        `Unable to send mail. No agency config found for ${convention.agencyId}`,
+      );
+    }
+
+    const {
+      beneficiary,
+      establishmentRepresentative,
+      beneficiaryCurrentEmployer,
+      beneficiaryRepresentative,
+    } = convention.signatories;
+
+    const recipients = [
+      beneficiary.email,
+      establishmentRepresentative.email,
+      ...agency.counsellorEmails,
+      ...agency.validatorEmails,
+    ];
+
+    if (beneficiaryCurrentEmployer && beneficiaryCurrentEmployer.email)
+      recipients.push(beneficiaryCurrentEmployer.email);
+    if (beneficiaryRepresentative && beneficiaryRepresentative.email)
+      recipients.push(beneficiaryRepresentative.email);
```

Comment:
Plutôt que de muter le tableau, tu peux faire tout dans la création du tableau (et le rendre unique au passage en utilisant ramda) :

```suggestion
    const recipients = uniq([
      beneficiary.email,
      establishmentRepresentative.email,
      ...(beneficiaryCurrentEmployer ? [beneficiaryCurrentEmployer.email] : [],
      ...(beneficiaryRepresentative ? [beneficiaryRepresentative.email] : [],
      ...agency.counsellorEmails,
      ...agency.validatorEmails,
    ]);
```